### PR TITLE
Pull Request from issue #2, where fetchPosts, isPost and isStringArray functions were created

### PR DESCRIPTION
**Pull request by OwenTruong**
fetchPosts is an async function that fetches post data from our database/backend, but right now it is just a placeholder (it does check types before returning the Post array though).

isPost checks if an unknown is of type Post.

isStringArray checks if an unknown is of type String[].

### CODE REVIEW

File: website/src/utils/isPost.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { isStringArray } from './isStringArray';
```
Comment (rollie42) [inline website/src/utils/isPost.ts:1]: I think libraries like Zod can help better structure this - rolling your own, it's easy to get out of sync with the model if you change, add, or remove a field.

Comment (OwenTruong) [reply website/src/utils/isPost.ts:1]: Thanks, I'll check out Zod! 

What do you mean by a field though? Do you mean as in the field of our interface, Post?

File: website/src/utils/isStringArray.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function isStringArray(obj: unknown): obj is string[] {
+  if (typeof obj === 'object' && Array.isArray(obj)) {
+    return obj.reduce(
+      (acc: boolean, el: unknown) => !(acc && typeof el != 'string'),
```
Comment (rollie42) [inline website/src/utils/isStringArray.ts:4]: This might be easier to read as `return !obj.find(el =>typeof el !== 'string')`

Comment (OwenTruong) [reply website/src/utils/isStringArray.ts:4]: That is true. It looks a lot cleaner! I'll make edits along with Zod before submitting another pull request. I'll reject this one.

**Pull request closed without merging**